packages/office-ui-fabric-react/src/components/Nav/Nav.scss

```diff
@@ -96,6 +96,11 @@ $linkRightPadding: 20px;
   vertical-align: middle;
 }
 
+.ms-Nav-linkHorizontal {
+  @include float(left);
+  display: inline;
```

Comment:
did you try inline-block? You shouldn't need float, and normal inline-block will do stacking like you'd expect. You may need to overflow clip things.